pako/index.d.ts

```diff
@@ -1,61 +1,84 @@
-// Type definitions for pako 0.2.8
+// Type definitions for pako 1.0.4
 // Project: https://github.com/nodeca/pako
-// Definitions by: Denis Cappellin <http://github.com/cappellin>
+// Definitions by: Denis Cappellin <https://github.com/cappellin>, Caleb Eggensperger <https://github.com/calebegg>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 export = Pako;
 export as namespace pako;
 
 declare namespace Pako {
+	export interface DeflateOptions {
+		level?: number;
+		windowBits?: number;
+		memLevel?: number;
+		strategy?: number;
+		dictionary?: any;
+		raw?: boolean;
+		to?: 'string';
+	}
+
+  export interface InflateOptions {
+		windowBits?: number;
+		raw?: boolean;
+		to?: 'string';
+	}
+
+	export type Data = Uint8Array | Array<number> | string;
+
 	/**
 	 * Compress data with deflate algorithm and options.
 	 */
-	export function deflate( data: Uint8Array | Array<number> | string, options?: any ): string;
+	export function deflate( data: Data, options: DeflateOptions & {to: 'string'} ): string;
+	export function deflate( data: Data, options?: DeflateOptions ): Uint8Array;
```

Comment:
I guess you are trying to enforce `to` to be a required property if the `options` is provided, however the optional parameter here ignores that requirement, and TypeScript do not differentiate signatures just via return types. Removing the `options?` here might work